daemon/k8s/script_test.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package k8s
+
+import (
+	"context"
+	"maps"
+	"os"
+	"testing"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/cilium/hive/script"
+	"github.com/cilium/hive/script/scripttest"
+	"github.com/cilium/statedb"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// TestScript runs all the testdata/*.txtar script tests. The tests are
+// run in parallel. If you need to update the expected files inside the txtar
+// files you can run 'go test . -scripttest.update' to update the files.
+func TestScript(t *testing.T) {
+	time.Now = func() time.Time {
+		return time.Date(2000, 1, 1, 10, 30, 0, 0, time.UTC)
+	}
+	os.Setenv("TZ", "")
```

Comment:
These should probably be restored after the test finishes, so they don't influence any successive tests:

```go
now := time.Now
time.Now = func() time.Time {
	return time.Date(2000, 1, 1, 10, 30, 0, 0, time.UTC)
}
defer func() { time.Now = now }()
t.Setenv("TZ", "")
```

Reply:
https://github.com/cilium/cilium/pull/37299